### PR TITLE
Make MapHub and MapEndpoint consistent with other ASP.NET Core APIs (use PathString) #1188

### PR DESCRIPTION
**Pull request by vany0114**
Hi @anurse 

This PR Makes MapHub and MapEndpoint consistent with other ASP.NET Core APIs using PathString. #1188

Thanks for your help!

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR/HubRouteBuilder.cs [hunk 1]
```diff
@@ -19,10 +20,10 @@ public HubRouteBuilder(SocketRouteBuilder routes)
 
         public void MapHub<THub>(string path) where THub : Hub
         {
-            MapHub<THub>(path, socketOptions: null);
+            MapHub<THub>(new PathString(path), socketOptions: null);
         }
 
-        public void MapHub<THub>(string path, Action<HttpSocketOptions> socketOptions) where THub : Hub
+        public void MapHub<THub>(PathString path, Action<HttpSocketOptions> socketOptions) where THub : Hub
```
Comment (anurse) [inline src/Microsoft.AspNetCore.SignalR/HubRouteBuilder.cs]: Let's also have an overload that just takes just a `PathString` (setting `socketOptions` to null).

File: src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs [hunk 2]
```diff
@@ -20,9 +21,9 @@ public SocketRouteBuilder(RouteBuilder routes, HttpConnectionDispatcher dispatch
         }
 
         public void MapSocket(string path, Action<ISocketBuilder> socketConfig) =>
-            MapSocket(path, new HttpSocketOptions(), socketConfig);
+            MapSocket(new PathString(path), new HttpSocketOptions(), socketConfig);
 
-        public void MapSocket(string path, HttpSocketOptions options, Action<ISocketBuilder> socketConfig)
+        public void MapSocket(PathString path, HttpSocketOptions options, Action<ISocketBuilder> socketConfig)
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs]: Same here, add an overload with just `(PathString path, Action<ISocketBuilder> socketConfig)`

File: src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs [hunk 3]
```diff
@@ -33,10 +34,10 @@ public void MapSocket(string path, HttpSocketOptions options, Action<ISocketBuil
 
         public void MapEndPoint<TEndPoint>(string path) where TEndPoint : EndPoint
         {
-            MapEndPoint<TEndPoint>(path, socketOptions: null);
+            MapEndPoint<TEndPoint>(new PathString(path), socketOptions: null);
         }
 
-        public void MapEndPoint<TEndPoint>(string path, Action<HttpSocketOptions> socketOptions) where TEndPoint : EndPoint
+        public void MapEndPoint<TEndPoint>(PathString path, Action<HttpSocketOptions> socketOptions) where TEndPoint : EndPoint
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs]: And again, an overload with just `(PathString path)`.

**Comment by vany0114** [comment]
Hey @anurse!

Sorry for the delay, today was a busy day, I just made the changes! ...BTW I made the changes one hour ago but the build was queued and now it failed because of a timeout in some unit test. (locally all tests passed) Should I push another change to trigger the build again?

**Comment by anurse** [comment]
No problem @vany0114 ! The tests are a bit inconsistent right now, we're working on fixing that. Until then, I can restart the build and hopefully it will pass. If not, I can look in to it further.

**Comment by anurse** [comment]
31ef3c49df90731169b9c8ea5a6a707964a8001c

**Pull request closed without merging**